middlewares/validators/monitor.js

```diff
@@ -0,0 +1,129 @@
+const joi = require("joi");
+const { VALID_PROGRESS_TYPES, TYPE_MAP } = require("../../constants/progresses");
+
+const baseSchema = joi
+  .object()
+  .strict()
+  .keys({
+    type: joi
+      .string()
+      .trim()
+      .valid(...VALID_PROGRESS_TYPES)
+      .required()
+      .messages({
+        "any.required": "Required field 'type' is missing.",
+        "any.only": "Type field is restricted to either 'user' or 'task'.",
+      }),
+    monitored: joi.boolean().required().messages({
+      "any.required": "Required field 'monitored' is missing.",
+      "boolean.base": "monitored field must be a boolean value.",
+    }),
+    frequency: joi
+      .number()
+      .integer()
+      .positive()
+      .when("type", {
+        is: "user",
+        then: joi.number().equal(1).messages({
+          "number.equal": "'frequency' field must be equal to 1",
+        }),
+        otherwise: joi.optional(),
+      })
+      .messages({
+        "number.base": "'frequency' field must be a number",
+        "number.integer": "'frequency' field must be an integer",
+        "number.positive": "'frequency' field must be a positive integer",
+        "any.only": "'frequency' field must be equal to 1 for type 'user'",
+      }),
+    taskId: joi
+      .string()
+      .trim()
+      .when("type", {
+        is: "task",
+        then: joi.required().messages({
+          "any.required": "Required field 'taskId' is missing.",
+          "string.trim": "taskId must not have leading or trailing whitespace",
```

Comment:
should we show this error? cant we trim trailing spaces in backend?

Reply:
can be handled at backend.